lib/app_jwt/app_jwt.c

```diff
@@ -0,0 +1,742 @@
+			     char *uuid_str_out, const int32_t uuid_str_out_size)
+{
+
+	if ((NULL == uuid_words) || (uuid_byte_len < UUID_BINARY_WORD_SZ * 4) ||
+	    (NULL == uuid_str_out) || (uuid_str_out_size < APP_JWT_UUID_V4_STR_LEN)) {
+		/* bad parameter */
+		return -EINVAL;
+	}
+	/* this will hold 4 integer words in string format (string Length 8 + 1) */
+	char temp_str[UUID_BINARY_WORD_SZ][BINARY_WORD_STR_SZ] = {0};
+
+	/* Transform integers to strings */
+	for (int i = 0; i < UUID_BINARY_WORD_SZ; i++) {
+		/* UUID byte endiannes is little endian first. */
+		sprintf(temp_str[i], "%08x", sys_cpu_to_be32(uuid_words[i]));
+	}
+
+	/* UUID string format defined by  ITU-T X.667 | ISO/IEC 9834-8 : */
+	/* <8 char>-<4 char>-<4 char>-<4 char>-<12 char> */
+
+	/* for some reason, sprintf will not respect the amout of characters to print that is */
+	/* specified in the formatting string. use this cascade of sprintf to overcome this issue.
+	 */
+	/* 59fa2510\0 , 8952abc5\0 , a20cd981\0 , 2bf2e936\0 */
+	/*'^         ' = "59fa2510" */
+	sprintf(uuid_str_out, "%8s", temp_str[0]);
+	/* 59fa2510\0 , 8952abc5\0 , a20cd981\0 , 2bf2e936\0 */
+	/*			   '^   ' = "8952" */
+	sprintf(uuid_str_out + 8, "-%4s", temp_str[1] + 0);
+	/* 59fa2510\0 , 8952abc5\0 , a20cd981\0 , 2bf2e936\0 */
+	/*			       '^   ' = "abc5" */
+	sprintf(uuid_str_out + 8 + 5, "-%4s", temp_str[1] + 4);
+	/* 59fa2510\0 , 8952abc5\0 , a20cd981\0 , 2bf2e936\0 */
+	/*  			            '^   ' = "a20c" */
+	sprintf(uuid_str_out + 8 + 5 + 5, "-%4s", temp_str[2] + 0);
+	/* 59fa2510\0 , 8952abc5\0 , a20cd981\0 , 2bf2e936\0 */
+	/*  			                '^   ' + '^         '  = "d9812bf2e936" */
+	sprintf(uuid_str_out + 8 + 5 + 5 + 5, "-%4s%8s", temp_str[2] + 4, temp_str[3]);
+	uuid_str_out[APP_JWT_UUID_V4_STR_LEN] = '\0';
+
+	return 0;
+}
+
+#if defined(CONFIG_APP_JWT_PRINT_EXPORTED_PUBKEY_DER)
+//---------------------------------------------------------------------------//
+static int bytes_to_str(const uint8_t *const bytes_buf, const size_t bytes_len, char *string_buff,
+			const size_t string_buff_size, size_t *string_len)
+{
+	int err = 0;
+
+	if ((NULL != bytes_buf) && (0 != bytes_len) && (NULL != string_buff) &&
+	    (0 != string_buff_size)) {
+		uint32_t str_pos = 0;
+		uint8_t byte_h, byte_l = 0;
+
+		for (uint32_t count = 0; count < bytes_len; count++) {
+			byte_h = ((bytes_buf[count] >> 4) & 0xF);
+			byte_l = (bytes_buf[count] & 0x0F);
+			if (byte_h < 10) {
+				/* transpose from 0-9 integer numbers */
+				/* to 0x30('0') - 0x31('9') ASCII characters */
+				byte_h += '0';
+			} else {
+				/* transpose from 10-15 integer numbers */
+				/* to 0x61('a') - 0x66('f') ASCII characters */
+				byte_h += ('a' - 10);
+			}
+
+			if (byte_l < 10) {
+				byte_l += '0';
+			} else {
+				byte_l += ('a' - 10);
+			}
+
+			sprintf(string_buff + str_pos, "%c%c", (char)(byte_h), (char)(byte_l));
+			str_pos += 2;
+		}
+
+		string_buff[str_pos] = '\0';
+		*string_len = str_pos + 1;
```

Comment:
Should probably make sure that string_len pointer is not NULL before dereference?